### PR TITLE
[chore][receiver/mysql] Migrate away from timing based setup in integration test

### LINKED ISSUE #18286: [receiver/mysql] Flaky integration test
### Component(s)

receiver/mysql

### What happened?

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4089471343/jobs/7052232713#step:6:544

```
    integration_test.go:70: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mysqlreceiver/integration_test.go:70
        	Error:      	Received unexpected error:
        	            	resource "map[mysql.instance.endpoint:localhost:3306]": scope "otelcol/mysqlreceiver": number of metrics doesn't match expected: 21, actual: 20
        	Test:       	TestMySqlIntegration/Running_mysql_version_8.0
--- FAIL: TestMySqlIntegration (0.00s)
    --- FAIL: TestMySqlIntegration/Running_mysql_version_8.0 (31.22s)
FAIL
```

origin from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18284

### Collector version

N/A

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by djaglowski**
Resolves #18286 

This moves away from a custom docker file w/ a timing based setup script. Instead, we rely on `testcontainers` to run a script once the container is ready.

Several minor accompanying changes were made here as well:
- Do not grant select permissions to the innodb_metrics table, as it turns out that each user automatically has appropriate row-based permissions to this table. Also remove documentation citing this as a prereq.
- Grant permissions to `performance_schema` tables, because we were logging errors from failed queries added in #14328.
- Do not consider a lack of replication metrics as an error, since these are expected to be absent unless replication is configured.
- Add a debugging feature to `scraperinttest`, which allows us to force a failure if any errors are logged by a scraper. This should only be used for developing tests due to a lack or correlation with individual scrapes.

### CODE REVIEW

**Comment by djaglowski** [comment]
@open-telemetry/collector-contrib-approvers, PTAL

File: receiver/mysqlreceiver/testdata/integration/scripts/setup.sh [hunk 1]
```diff
@@ -1,46 +0,0 @@
...
-
-
-setup_permissions() {
-    # NOTE: -pPASSWORD is missing a space on purpose
-    mysql -u root -p"${ROOT_PASS}" -e "GRANT PROCESS ON *.* TO ${USER}" > /dev/null
-    mysql -u root -p"${ROOT_PASS}" -e "GRANT SELECT ON INFORMATION_SCHEMA.INNODB_METRICS TO ${USER}" > /dev/null
-    mysql -u root -p"${ROOT_PASS}" -e "FLUSH PRIVILEGES" > /dev/null
-}
-
-setup_data() {
```
Comment (evan-bradley) [inline receiver/mysqlreceiver/testdata/integration/scripts/setup.sh:20]: It doesn't look like these were replaced anywhere, are they no longer needed?

**Review by evan-bradley — APPROVED** [review]
Looks good to me, just one question.

File: receiver/mysqlreceiver/testdata/integration/scripts/setup.sh (hunk 1 above)
Comment (djaglowski) [reply receiver/mysqlreceiver/testdata/integration/scripts/setup.sh:20]: Yeah, they appear to have been extraneous.

**Review by fatsheep9146 — APPROVED** [review]

**Pull request merged**